Mathlib/Data/Fin/Basic.lean

```diff
@@ -1654,13 +1654,24 @@ theorem succRecOn_succ {C : ∀ n, Fin n → Sort _} {H0 Hs} {n} (i : Fin n) :
     @Fin.succRecOn (n + 1) i.succ C H0 Hs = Hs n i (Fin.succRecOn i H0 Hs) := by cases i; rfl
 #align fin.succ_rec_on_succ Fin.succRecOn_succ
 
+-- porting note: This is a bandaid solution to make Fin.induction computable
+-- This can (and probably should) be removed once the code generator supports `Nat.rec`
+
+/-- A computable definition for `Fin.induction`-/
+def inductionImpl {C : Fin (n + 1) → Sort _} (h0 : C 0)
+    (hs : ∀ i : Fin n, C (castSucc i) → C i.succ) :
+    ∀ i : Fin (n + 1), C i :=
+  fun ⟨i, hi⟩ => by cases i with
+  | zero => rwa [Fin.mk_zero]
+  | succ i => exact hs ⟨i, lt_of_succ_lt_succ hi⟩ (Fin.inductionImpl h0 hs ⟨i, lt_of_succ_lt hi⟩)
+
+--porting note: see `Fin.inductionImpl` above for added `implemented_by`
 /-- Define `C i` by induction on `i : Fin (n + 1)` via induction on the underlying `Nat` value.
 This function has two arguments: `h0` handles the base case on `C 0`,
 and `hs` defines the inductive step using `C i.castSucc`.
 -/
-@[elab_as_elim]
---Porting note: marked noncomputable
-noncomputable def induction {C : Fin (n + 1) → Sort _} (h0 : C 0)
+@[elab_as_elim, implemented_by Fin.inductionImpl]
```

Comment:
Don't you need a `csimp` lemma that proves these are the same?

Reply:
I didn't know what that was, sorry. But I've just added it. Please let me know if there's anything I need to add on top.